### ISSUE TITLE
Allow assigns transformation before render for Components

### ISSUE THREAD

**Issue by sezaru** [surface-ui/surface#622]
Imagine a scenario where I send a structure to a `prop` of a component but that prop needs to be manipulated someway before being shown to the user, let's say I need to filter it and then count.

Currently, I can do that if I use the `render/1` function like this:

```elixir
def render(assigns) do
  count = assigns.data |> Enum.filter(...) |> Enum.count()

  ~F"""
  <div>count: {count}</div>
  """
end
```

The issue here is that if I want to use a `sface` file instead, I simply cannot do the same assign prop transformation anymore.

Would be possible to add an `update` function that would receive the assigns and return new assigns in `Surface.Component`?
Or is there any way to call the `sface` render inside the component render function?

**Comment by simonmcconnell** [comment]
On a side note, the way you're doing it there is incorrect. You need to use `assigns = assign(assigns, :count, ...)` in `render` and use `@count` instead of creating a variable and referencing the bare variable inside the template. This keeps change tracking happy.

**Comment by sezaru** [comment]
Thanks for the tip, I was not aware of that!

**Comment by Malian** [comment]
@sezaru the solution proposed by @simonmcconnell is the way to go.

We used an `update` callback few version ago, but it is not consistent with what Phoenix Live View does, so we removed it. See the migration section for more information:

> Surface v0.6.x relies on the Liveview features available since v0.16. The main change from the user perspective is that the stateless Surface.Component now is built on top of Phoenix.Component instead of Phoenix.LiveComponent. This means the mount/1, preload/1 and update/2 callbacks are no longer available. 

Currently, having a `render` method and a `.sface` file will result in a warning message, and the `.sface` file will be ignored.

That being said, I understand the need to "update" some `assigns` while having a colocated `sface` template. One solution could be to add an helper method to render a `.sface` file manually from the `render` function like:

```elixir
def render(assigns) do
  count = assigns(assigns, :count, 3)
  Surface.render_sface(filepath)
end
```

I am still wondering how to remove the warning.

@msaraiva @davydog187 @miguel-s @tiagoefmoraes What do you think?

**Comment by sezaru** [comment]
Would it be hard to add that helper function? I think it would be kind of similar to Phoenix colocating templates correct?

**Comment by msaraiva** [comment]
**Update:** This has been already addressed on this branch: https://github.com/surface-ui/surface/tree/new-context-api.

Usage as suggested:

```
  def render(assigns) do
    assigns = assign(assigns, :value, 123)
    render_sface(assigns)
  end
```

**Comment by msaraiva** [comment]
Fixed in #641.
